src/Template.php

```diff
@@ -221,8 +221,7 @@ public function displayBlock($name, array $context, array $blocks = [], $useBloc
 
                 // this is mostly useful for \Twig\Error\LoaderError exceptions
                 // see \Twig\Error\LoaderError
-                if (-2 === $e->getTemplateLine()) {
-                    $e->setTemplateLine(-1);
+                if (-1 === $e->getTemplateLine()) {
```

Comment:
couldn't we avoid this line guessing in most cases by passing the appropriate location to the `displayBlock` method when compiling a BlockReferenceNode instead ?

Reply:
sure, and that's more or less what we are already doing in `Template::loadTemplate()`.